py/picca/pk1d/postproc_pk1d.py

```diff
@@ -749,22 +752,30 @@ def compute_average_pk_redshift(
                     standard_dev**2,
                     bounds=(0, np.inf),
                 )
+                coef_error, *_ = curve_fit(
+                    fitfunc_variance_pk1d,
+                    snr_bins,
+                    standard_dev_error**2,
+                    bounds=(0, np.inf),
+                )
 
                 # Model variance from fit function
                 data_snr[data_snr > MEANPK_FITRANGE_SNR[1]] = MEANPK_FITRANGE_SNR[1]
                 data_snr[data_snr < 1.01] = 1.01
                 variance_estimated = fitfunc_variance_pk1d(data_snr, *coef)
+                variance_estimated_error = fitfunc_variance_pk1d(data_snr, *coef_error)
                 weights = 1.0 / variance_estimated
+                weights_error = 1.0 / variance_estimated_error
                 if apply_z_weights:
                     weights *= redshift_weights
                 mean = np.average(data_values, weights=weights)
                 if apply_z_weights:
                     # Analytic expression for the re-weighted average:
-                    error = np.sqrt(np.sum(weights * redshift_weights)) / np.sum(
-                        weights
+                    error = np.sqrt(np.sum(weights_error * redshift_weights)) / np.sum(
```

Comment:
why would you want to use different weights for the mean than you'd use for the standard deviation?

I.e. for the mean you just weigh by the varance estimated for the p1d measurement, but for the std you weigh by the variance for that particular column which would be internally inconsistent.
When you say `error` here, should this be the error on the mean? Or the standard deviation of the dataset around the mean

Reply:
See answer below for detail on that